cni/pkg/nodeagent/server.go

```diff
@@ -226,17 +278,23 @@ func (s *meshDataplane) Stop(skipCleanup bool) {
 	if !skipCleanup {
 		log.Info("CNI ambient server terminating, cleaning up node net rules")
 
-		log.Debug("removing host iptables rules")
-		s.hostIptables.DeleteHostRules()
+		_ = runAsNs(s.hostNetNS, func() error {
+			log.Debug("removing host iptables rules")
+			s.hostIptables.DeleteHostRules()
```

Comment:
ditto - `host` funcs should obscure all `host` namespace jumps (or they aren't much use)